.github/workflows/functional_test_v2.yaml

```diff
@@ -488,3 +488,270 @@ jobs:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Checkout old release
+        uses: actions/checkout@v4
+        with:
+          path: base
+          ref: splunk-otel-collector-0.113.0
+          fetch-depth: 0
+      - uses: actions/setup-go@v5
+        with:
+          go-version: ~1.22.5
+          cache: false
+      - name: Cache Go
+        id: go-cache
+        timeout-minutes: 5
+        uses: actions/cache@v4
+        with:
+          path: |
+            ~/go/bin
+            ~/go/pkg/mod
+          key: go-cache-${{ runner.os }}-${{ hashFiles('**/go.sum') }}
+      - name: Create kind cluster
+        uses: helm/kind-action@v1.12.0
+        with:
+          node_image: kindest/node:v1.30.0
+          kubectl_version: v1.30.0
+          cluster_name: kind
+          config: ./.github/workflows/configs/kind-config.yaml
+      - name: Fix kubelet TLS server certificates
+        run: |
+          kubectl get csr -o=jsonpath='{range.items[?(@.spec.signerName=="kubernetes.io/kubelet-serving")]}{.metadata.name}{" "}{end}' | xargs kubectl certificate approve
+      - name: Update dependencies
+        run: |
+          cd base && make dep-update
+      - name: Deploy cert-manager
+        run: |
+          cd base && make cert-manager
+      - name: install helm chart with operator enabled
+        run: |
+          cd base && helm install sock helm-charts/splunk-otel-collector --debug \
+          --set splunkObservability.realm=us0 --set splunkObservability.accessToken=xxxxx \
+          --set operator.enabled=true --set environment=dev --set clusterName=dev
+        continue-on-error: true
+      - name: check crds are installed
+        run: |
+          kubectl get crds | grep 'opentelemetry.io'
+
+          echo "get instrumentations - none if install failed"
+          kubectl get instrumentations.opentelemetry.io -A
+      - name: check operator
+        run: |
+          kubectl describe po -lapp.kubernetes.io/name=operator
+          kubectl get po -lapp.kubernetes.io/name=operator -o=name | xargs -n 1 kubectl wait --for=condition=Ready --timeout=2m
+          kubectl describe po -lapp.kubernetes.io/name=operator
+        continue-on-error: true
+      - name: upgrade old again
+        run: |
+          cd base && helm upgrade sock helm-charts/splunk-otel-collector --debug \
+          --set splunkObservability.realm=us0 --set splunkObservability.accessToken=xxxxx \
+          --set operator.enabled=true --set environment=dev --set clusterName=dev
+        continue-on-error: true
+      - name: check helm release status
+        run: |
+          helm status sock
+      - name: check instrumentations installed
+        run: |
+          kubectl describe instrumentations.opentelemetry.io -A
+      - name: Update dependencies
+        run: |
+          make dep-update
+      - name: try to import instrumentation in helm
```

Comment:
required cmds to get around the instrumentation hook to non-hook change on upgrade